pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JavaQualifiedName.java

```diff
@@ -165,13 +287,27 @@ public static JavaQualifiedName ofString(String name) {
             return null;
         }
 
-        qname.packages = ".".equals(matcher.group(1)) ? null : matcher.group(1).split("\\.");
-        qname.classes = matcher.group(3).split("\\$");
-        qname.operation = matcher.group(6) == null ? null : matcher.group(6).substring(1);
+        qname.packages = ".".equals(matcher.group(PACKAGES_GROUP_INDEX)) ? null : matcher.group(PACKAGES_GROUP_INDEX).split("\\.");
+        qname.operation = matcher.group(OPERATION_GROUP_INDEX) == null ? null : matcher.group(OPERATION_GROUP_INDEX).substring(1);
+
+        qname.classes = matcher.group(CLASSES_GROUP_INDEX).split("\\$");
+        qname.localIndices = new int[qname.classes.length];
+        Pattern localIndexPattern = Pattern.compile("(\\d+)(\\w+)");
```

Comment:
any reason to not have this pattern as a static final to avoid recompiling the regexp each time?

Reply:
No reason, except for not polluting the field namespace with that local artifact. I'll change it